### PR TITLE
Mavlink v2.0 specs for RC_CHANNELS_OVERRIDE accepts upto 18 channels.…

### PR DESCRIPTION
**Pull request by amilcarlucas**
… The plugin publishes channels 9 to 18 if the FCU protocol version is 2.0

### CODE REVIEW

File: mavros/src/plugins/rc_io.cpp [hunk 1]
```diff
@@ -247,6 +247,18 @@ class RCIOPlugin : public plugin::PluginBase {
...
+			ovr.chan10_raw = req->channels[9];
+			ovr.chan11_raw = req->channels[10];
+			ovr.chan12_raw = req->channels[11];
+			ovr.chan13_raw = req->channels[12];
+			ovr.chan14_raw = req->channels[13];
+			ovr.chan15_raw = req->channels[14];
+			ovr.chan16_raw = req->channels[15];
+			ovr.chan17_raw = req->channels[16];
+			ovr.chan18_raw = req->channels[17];
+		}
```
Comment (vooon) [inline mavros/src/plugins/rc_io.cpp:261]: 1. That code should be generated by cog.
2. With change in message type you'll switching from std::array to std::vector (or boost:: for ros1), so you must verify size of `channels`.

**Review by vooon — CHANGES_REQUESTED** [review]
You must check vector size to prevent sigfaults.

I also do not like that old `chan%d_raw` style messages, because you always need to write a lot of loc.
Can we do something like that (pseudocode)
```
auto mapper = std::array<*uint16_t, 18>(&ovr.chan1_raw ...);
std::copy(req->channels.cbegin(), req->channels.cend(), mapper.begin());
```

**Comment by vooon** [comment]
NOTE: you don't need to check that lint is in v2.0 mode, just fill in fields, mavlink library would drop them automatically.

**Comment by amilcarlucas** [comment]
@karthikdesai can you please follow up on this?

**Comment by karthikdesai** [comment]
@vooon 
cool thanks for the suggestion. i had actually opened up an issue for this and you had suggested me to use the cog script at the time. https://github.com/mavlink/mavros/issues/1383
The new commit @amilcarlucas pushed actually are the changes what i did after i followed on your comment on the issue ^^

With regards to the message type, i had defined the array size as 18 at the time so it is not a vector but an array itself, so the verification wasn't needed

**Review by vooon — APPROVED** [review]
LGTM, Thanks!

**Comment by vooon** [comment]
Fix #1338.

**Pull request merged**